api/types.go

```diff
@@ -86,11 +86,28 @@ type DataSourceInfo struct {
 	Parameters       map[string]string `json:"parameters"`
 	ExpectedChecksum string            `json:"expectedChecksum"`
 
-	FileName        string `json:"fileName"`
-	State           string `json:"state"`
-	Size            int64  `json:"size"`
-	Progress        int    `json:"progress"`
-	ProcessedSize   int64  `json:"processedSize"`
-	CurrentChecksum string `json:"currentChecksum"`
-	Message         string `json:"message"`
+	FilePath         string `json:"filePath"`
+	State            string `json:"state"`
+	Size             int64  `json:"size"`
+	Progress         int    `json:"progress"`
+	ProcessedSize    int64  `json:"processedSize"`
+	ModificationTime string `json:"modificationTime"`
+	CurrentChecksum  string `json:"currentChecksum"`
+	Message          string `json:"message"`
+}
+
+type FileInfo struct {
```

Comment:
FIleInfo is the same as DataSourceInfo? Do you consider using embedded structure or alias?